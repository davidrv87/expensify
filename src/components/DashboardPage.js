import React from 'react';

const DashboardPage = () => (
    <div>
        This is the Dashboard Page
    </div>
);

export default DashboardPage;
