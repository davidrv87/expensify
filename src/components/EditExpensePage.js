import React from 'react';

const EditExpensePage = props => (
    <div>
        This is the edit expense page for id {props.match.params.id}
    </div>
);

export default EditExpensePage;
