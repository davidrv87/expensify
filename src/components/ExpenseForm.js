import React from 'react';
import { connect } from 'react-redux';
import moment from 'moment';
import 'react-dates/initialize'; // Required as per v13 - see https://github.com/airbnb/react-dates/issues/797
import { SingleDatePicker } from 'react-dates';
import 'react-dates/lib/css/_datepicker.css';

import { addExpense } from "../actions/expenses";

class ExpenseForm extends React.Component {
    state = {
        description: '',
        amount: '',
        createdAt: moment(),
        calendarFocused: false,
        note: ''
    }
    onDescriptionChange = (e) => {
        const description = e.target.value;
        this.setState(() => ({ description }));
    };
    onNoteChange = (e) => {
        const note = e.target.value;
        this.setState(() => ({ note }));
    };
    onAmountChange = (e) => {
        const amount = e.target.value;
        if (!amount || amount.match(/^\d+(\.\d{0,2})?$/)) {
            this.setState(() => ({ amount }));
        }
    };
    onDateChange = (createdAt) => {
        if (createdAt) {
            this.setState(() => ({ createdAt }));
        }
    };
    onFocusChange = ({ focused }) => {
        this.setState(() => ({ calendarFocused: focused }));
    };
    onFormSubmit = (e) => {
        e.preventDefault();

        this.props.dispatch(addExpense(this.state));
    };
    render() {
        return (
            <div>
                <form
                    onSubmit={this.onFormSubmit}
                >
                    <input
                        type="text"
                        placeholder="Description"
                        autoFocus
                        value={this.state.description}
                        onChange={this.onDescriptionChange}
                    />
                    <input
                        type="text"
                        placeholder="Amount"
                        value={this.state.amount}
                        onChange={this.onAmountChange}
                    />
                    <SingleDatePicker
                        date={this.state.createdAt}
                        onDateChange={this.onDateChange}
                        focused={this.state.calendarFocused}
                        onFocusChange={this.onFocusChange}
                        numberOfMonths={1}
                        isOutsideRange={(day) => false} // Enable past days
                    />
                    <textarea
                        placeholder="Add a note for your expense (optional)"
                        value={this.state.note}
                        onChange={this.onNoteChange}
                    />
                    <button>Create Expense</button>
                </form>
            </div>
        );
    }
}

export default connect()(ExpenseForm);
