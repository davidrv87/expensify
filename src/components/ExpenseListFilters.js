import React from 'react';
import { connect } from 'react-redux';

import { setTextFilter, sortByAmount, sortByDate } from '../actions/filters';

const ExpenseListFilters = props => (
    <div>
        <input
            type="text"
            value={props.filters.text}
            onChange={(e) => {
                // When we connect a Component, we have access to dispatch() via props.dispatch
                // This can be seen using the React Dev Tool in Chrome
                props.dispatch(setTextFilter(e.target.value));
            }}
        />
        <select
            value={props.filters.sortBy}
            onChange={(e) => {
                e.target.value === 'date' ? props.dispatch(sortByDate()) : props.dispatch(sortByAmount());
            }}
        >
            <option value="date">Date</option>
            <option value="amount">Amount</option>
        </select>
    </div>
);

const mapStateToProps = state => ({
    filters: state.filters
});

export default connect(mapStateToProps)(ExpenseListFilters);
