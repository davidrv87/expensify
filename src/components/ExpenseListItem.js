import React from 'react';
import { connect } from 'react-redux';

import { removeExpense } from '../actions/expenses';

const ExpenseListItem = ({ id, description, amount = 0, createdAt = 0, dispatch }) => (
    <div>
        <h3>{description}</h3>
        <p>Amount: {amount}</p>
        <p>Created: {createdAt}</p>
        <button
            onClick={(e) => {
                dispatch(removeExpense({ id }));
            }}
        >
            Remove
        </button>
    </div>
);

export default connect()(ExpenseListItem);
