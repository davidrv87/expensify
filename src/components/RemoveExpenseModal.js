import React from 'react';
import Modal from 'react-modal';

const RemoveExpenseModal = props => (
    <Modal
        isOpen={!!props.expenseToRemove}
        contentLabel="Confirm Remove Expense"
        onRequestClose={props.handleCancelRemove}
        appElement={document.getElementById('app')}
        closeTimeoutMS={200}
        className="modal"
    >
        <h3 className="modal__title">Remove Expense</h3>
        {props.expenseToRemove && <p className="modal__body">{props.expenseToRemove}</p>}

        <div className="modal__buttons">
            <button className="button" onClick={props.handleConfirmRemove}>Confirm</button>
            <button className="button button--secondary" onClick={props.handleCancelRemove}>Cancel</button>
        </div>
    </Modal>
);

export default RemoveExpenseModal;