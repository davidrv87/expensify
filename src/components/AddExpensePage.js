import React from 'react';
import { connect } from 'react-redux';

import ExpenseForm from './ExpenseForm';
import { addExpense } from '../actions/expenses';


const AddExpensePage = props => (
    <div>
        <h1>Create Expense</h1>
        <ExpenseForm
            onSubmit={(expense) => {
                props.dispatch(addExpense(expense));
                // This can be seen in the 'React' tab in DevTools
                props.history.push('/');
            }}
        />
    </div>
);

export default connect()(AddExpensePage);
