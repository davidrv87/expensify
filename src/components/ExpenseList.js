import React from 'react';
import { connect } from 'react-redux';

import ExpenseListItem from './ExpenseListItem';
import selectExpenses from '../selectors/expenses';

// Unconnected component
const ExpenseList = props => (
    <div>
        <h1>Expense List</h1>
        {
            props.expenses.map(expense => (
                <ExpenseListItem key={expense.id} {...expense} />
            ))
        }
    </div>
);

// Define the things we want to get from the store
const mapStateToProps = state => ({
    expenses: selectExpenses(state.expenses, state.filters)
});

// Connected component
export default connect(mapStateToProps)(ExpenseList);
