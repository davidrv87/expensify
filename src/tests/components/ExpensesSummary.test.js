import React from 'react';
import { shallow } from 'enzyme';

import { ExpensesSummary } from '../../components/ExpensesSummary';

test('should render ExpensesSummary correctly with 1 expense', () => {
    const wrapper = shallow(
        <ExpensesSummary
            expensesCount={1}
            expensesTotal={1234}
        />
    );
    expect(wrapper).toMatchSnapshot();
});

test('should render ExpensesSummary correctly with 2 expenses', () => {
    const wrapper = shallow(
        <ExpensesSummary
            expensesCount={2}
            expensesTotal={123444}
        />
    );
    expect(wrapper).toMatchSnapshot();
});
