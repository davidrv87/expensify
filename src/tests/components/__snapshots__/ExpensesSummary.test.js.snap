// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesSummary correctly with 1 expense 1`] = `
<div>
  <h2>
    Viewing 
    1
     
    expense
     totalling 
    $12.34
  </h2>
</div>
`;

exports[`should render ExpensesSummary correctly with 2 expenses 1`] = `
<div>
  <h2>
    Viewing 
    2
     
    expenses
     totalling 
    $1,234.44
  </h2>
</div>
`;
